### PR TITLE
ROR capstone project

### PR DESCRIPTION
**Pull request by omar25ahmed**
### Important note:
- I'm using SendGrid to have a real verification so when you sign up use your emails

# in this milestone:
- Use ruby gems as software packages system.
- Install Ruby on Rails framework.
- Understand Rails RESTful design and router.
- Use controllers to handle requests and render empty views.
- Use params from browser request in a safe way.
- Use preprocessed html file with embedded Ruby code.
- Use layouts and templates for shared content.
- Use database migration files to maintain database schema.
- Use validations for models.
- Secure app from n+1 problems.
- Understand what ORM is.
- Write SQL queries with ActiveRecord.
- Set up associations between models.
- Build a webapp that requires the user to log in.
- Use devise gem for authentication.
- Limit access to webapp resources based on authorization rules.
- Analyze in writing why you have made a coding choice using one structure over another.

## Project documentation:
https://www.loom.com/share/37c16ba85ab0437aae6ffbfd00145e27

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,64 @@
...
+
+👤 **Omar Ragheb**
+
+- GitHub: [https://github.com/omar25ahmed](https://github.com/omar25ahmed)
+- LinkedIn: [https://www.linkedin.com/in/omar-ragheb/](https://www.linkedin.com/in/omar-ragheb/)
+- Twitter: [https://twitter.com/Omar25Ahmed](https://twitter.com/Omar25Ahmed)
+
+## Show your support
+
+Give a ⭐️ if you like this project!
```
Comment (andersonlebon) [inline README.md:64]: - [ ] Kindly make sure your README contains an "Acknowledgements" section or similar that gives appropriate credit to [Gregoire Vella on Behance](https://www.behance.net/gregoirevella) the [author of the original design](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=), as required by the [Creative Commons license of the design](https://creativecommons.org/licenses/by-nc/4.0/).

File: app/views/categories/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+      <div class="card-body">
+        <h5 class="card-title"><%= fa_icon "#{category.icon}" %><%= category.name %></h5>
+        <p class="card-text">Total amount of transactions: <%= category.deals.map {|deal| deal.amount }.sum %>$</p>
+        <%= link_to "Show this category", category, class: "btn btn-primary"%>
+      </div>
+    </div>
+  <% end %>
+</div>
+
+<%= link_to "New category", new_category_path, class: "category-button"%>
```
Comment (andersonlebon) [inline app/views/categories/index.html.erb:16]: - [ ] Kindly note that the design requirements say that you should follow these [design guidelines](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=), including Colors, Typography: font face, size, and weight, Layout: composition and space between elements.
- Kindly make sure your category follows the template design layout

Your design | the template
---|---
image | image
- Make sure also a transaction category has an icon just like in the original template
- Note that also the new category button should always be at the bottom of the view(screen) even though there not a category in the database

Your design | the template 
---|---
image | image

**Review by andersonlebon — CHANGES_REQUESTED** [review]
## Hi @omar25ahmed 

Good job so far! I know you have worked hard on making this milestone But  There are some issues that you still need to work on to go to the next project but you are almost there!

almost-there
## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Your design looks good :100: :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
## Required Changes ♻️

- [ ] Kindly check comments under the review.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by MahmoudBakr23 — APPROVED** [review]
Hi @omar25ahmed  😃, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉

**Highlights**
- Well done making the suggested changes from the previous reviewer
- Your app is awesome. Keep up the great work 💯  

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**